packages/svelte/src/internal/server/hydratable.js

```diff
@@ -57,32 +57,28 @@ function encode(key, value, unresolved) {
 
 	entry.serialized = devalue.uneval(entry.value, (value, uneval) => {
 		if (is_promise(value)) {
+			// we serialize promises as `"${i}"`, because it's impossible for that string
+			// to occur 'naturally' (since the quote marks would have to be escaped)
+			// this placeholder is returned synchronously from `uneval`, which includes it in the
+			// serialized string. Later (at least one microtask from now), when `p.then` runs, it'll
+			// be replaced.
+			const placeholder = `"${uid++}"`;
 			const p = value
-				.then((v) => `r(${uneval(v)})`)
+				.then((v) => {
+					entry.serialized = entry.serialized.replace(placeholder, `r(${uneval(v)})`);
```

Comment:
As always all this recursive promise stuff is brain-bending, but I think we just had indirection here in the first place. There's no reason we need to actually return the string from this promise, we can just replace the placeholders inline.